ext/ManifoldsBaseRecursiveArrayToolsExt.jl

```diff
@@ -3,92 +3,72 @@ module ManifoldsBaseRecursiveArrayToolsExt
+        parallel_transport_to,
+        project,
+        _retract,
+        submanifold_component,
+        submanifold_components,
+        vector_transport_direction,
+        _vector_transport_direction,
+        _vector_transport_to,
+        vector_transport_to
+    import Base: copyto!, getindex, setindex!, view
 else
     # imports need to be relative for Requires.jl-based workflows:
     # https://github.com/JuliaArrays/ArrayInterface.jl/pull/387
     using ..ManifoldsBase
     using ..RecursiveArrayTools
-    using ..ManifoldsBase: AbstractBasis, VectorBundleBasisData
+    using ..ManifoldsBase:
+        AbstractBasis,
+        FiberBundle,
+        FiberBundleBasisData,
+        FiberBundleProductVectorTransport,
+        ProductBasisData,
+        TangentSpaceType
+    using Random
 
-    import ..ManifoldsBase: get_vectors, _vector_transport_to
-    import ..Base: getindex, setindex!, view
+    import ..ManifoldsBase:
+        allocate,
+        allocate_result,
+        default_inverse_retraction_method,
+        default_retraction_method,
+        default_vector_transport_method,
+        get_vector,
+        get_vectors,
+        inverse_retract,
+        parallel_transport_direction,
+        parallel_transport_to,
+        project,
+        _retract,
+        submanifold_component,
+        submanifold_components,
+        vector_transport_direction,
+        _vector_transport_direction,
+        _vector_transport_to,
+        vector_transport_to
+    import ..Base: copyto!, getindex, setindex!, view
 end
 
-function get_vectors(
-    M::VectorBundle,
-    p::ArrayPartition,
-    B::CachedBasis{𝔽,<:AbstractBasis{𝔽},<:VectorBundleBasisData},
-) where {𝔽}
-    xp1 = submanifold_component(p, Val(1))
-    zero_m = zero_vector(M.manifold, xp1)
-    zero_f = zero_vector(M.fiber, xp1)
-    vs = typeof(ArrayPartition(zero_m, zero_f))[]
-    for bv in get_vectors(M.manifold, xp1, B.data.base_basis)
-        push!(vs, ArrayPartition(bv, zero_f))
-    end
-    for bv in get_vectors(M.fiber, xp1, B.data.vec_basis)
-        push!(vs, ArrayPartition(zero_m, bv))
-    end
-    return vs
-end
-
-"""
-    getindex(p::ArrayPartition, M::VectorBundle, s::Symbol)
-    p[M::VectorBundle, s]
-
-Access the element(s) at index `s` of a point `p` on a [`VectorBundle`](@ref) `M` by
-using the symbols `:point` and `:vector` for the base and vector component, respectively.
-"""
-@inline function Base.getindex(p::ArrayPartition, M::VectorBundle, s::Symbol)
-    (s === :point) && return p.x[1]
-    (s === :vector) && return p.x[2]
-    return throw(DomainError(s, "unknown component $s on $M."))
-end
-
-"""
-    setindex!(p::ArrayPartition, val, M::VectorBundle, s::Symbol)
-    p[M::VectorBundle, s] = val
-
-Set the element(s) at index `s` of a point `p` on a [`VectorBundle`](@ref) `M` to `val` by
-using the symbols `:point` and `:vector` for the base and vector component, respectively.
+include("ProductManifoldRecursiveArrayToolsExt.jl")
```

Comment:
could we maybe either put these two or even all three files in a subfolder? For a short moment I thought these are now 3 extensions while it is only one. I think in Manifolds.jl we did a folder of the name of the Extension.

But cool that also product comes here!

Reply:
Sure, I will move them to a folder.

Reply:
I will also take the chance to review a few (currently a bit short) doc strings in the parts we move.

But I really like that with this PR – ManifoldsBase is: A nice interface for manifolds and a set of meta manifolds you can directly use as well. Super cool.

Reply:
Cool, thanks. I think moving those parts is a good idea.